### ISSUE TITLE
Tracks+Art view / Compact layout

### ISSUE THREAD

**Issue by StrangeSeasOfThought** [Taiko2k/TauonMusicBox#170]
**Tracks+Art:**

The frame (art box) is too large for album information to be displayed. All my artwork is 500x500 which is far from filling out the the frame (red dots).
So I've basically plenty of 'unused' space to display the album information.
I'd prefer both sidepanels (tracks+art & gallery) to have the same width.

Keep width of sidepanel, smaller art box (and ideally album information aligned to artwork).

![albuminformation](https://user-images.githubusercontent.com/43582772/64913017-85890280-d738-11e9-954c-cd31f422285c.png)

If you can point me in the right direction, I'd be fine with just changing this locally.

**Compact layout:**

- I changed the size of the biography artwork (compact size) from 200x200 to 150x150.
200x200 looks too large on the new compact layout. Going to compact layout removes biography, though. If it is enabled in normal layout, keep it enabled in compact layout, too. Just make the artist artwork smaller, 150x looks good in compact layout, imo:

![compact](https://user-images.githubusercontent.com/43582772/64913166-722b6680-d73b-11e9-8106-d793fb79af51.png)

- Only the first four or so playlist tabs are being displayed and I use the mouse wheel to scroll through all tabs (even the invisible ones). Is it possible to display the invisible tabs somehow? Maybe some kind of pop up info when scrolling?

**Comment by Taiko2k** [comment]
> 150x looks good in compact layout

Okay, I have made this change

> Keep width of sidepanel, smaller art box

Perhaps something like this:

![Screenshot from 2019-09-15 12-55-47](https://user-images.githubusercontent.com/17271572/64915347-75186c80-d7b8-11e9-8f32-10e5fd239790.png)

> Is it possible to display the invisible tabs somehow? Maybe some kind of pop up info when scrolling?

I agree there should at least be some feedback as to what playlist you've scrolled to. I'm not sure what the best way is though. A simple notification like pop-up of the name would be easiest.

**Comment by StrangeSeasOfThought** [comment]
> > Keep width of sidepanel, smaller art box
> 
> Perhaps something like this:

Looks good.

**Comment by Taiko2k** [comment]
> Plus album title, year, genre if possible.

Ive pushed my changes for this now. You can find this new setting under `Settings` > `View` > `Use centered style`

**Comment by StrangeSeasOfThought** [comment]
Perfect! Thanks!

**Comment by Taiko2k** [comment]
> the hide/show art box menu option is not working

Oh, yeah.

Okay fixed this now. Also allowed lyrics now by hiding the art as well.

**Comment by StrangeSeasOfThought** [comment]
Is there a reason for no showcase visualisation (compact layout)? Looks somewhat empty that way:

![showcase](https://user-images.githubusercontent.com/43582772/64923040-62ad2b80-d7d6-11e9-8ac4-a622876a125a.png)

**Comment by Taiko2k** [comment]
> Is there a reason for no showcase visualisation

No reason. Ive enabled it now.

> Track title font size should probably be somewhat smaller

Done. Also made the full size title font slightly smaller since I think it was a little oversize.

> Maybe some kind of pop up info when scrolling?

Done. Its not an ideal solution, but will do for now.

**Comment by StrangeSeasOfThought** [comment]
> > Maybe some kind of pop up info when scrolling?
> 
> Done. Its not an ideal solution, but will do for now.

Doesn't work (same for queue notification) when artist list view is visible. It's obviously not needed for the playlist/queue view but should still pop-up otherwise, imo.

**Comment by Taiko2k** [comment]
> Doesn't work (same for queue notification) when artist list view is visible

That *was* intentional, but you might be right, so I've changed it so those toasts always show now.

> a compact artist list for compact layout might be nice

Problem is though, its quite common for the list to be not fully populated with thumbnails. Even like only ~20% if the user hasn't configured anything further, so it would make the list very difficult to navigate without names always showing.

Though, having the side panel narrower at least is probably a good idea, so I've made that change:

![Screenshot from 2019-09-16 23-58-00](https://user-images.githubusercontent.com/17271572/64956831-a0e24200-d8df-11e9-9798-61329ae21b5d.png)

**Comment by StrangeSeasOfThought** [comment]
> Problem is though, its quite common for the list to be not fully populated with thumbnails. Even like only ~20% if the user hasn't configured anything further, so it would make the list very difficult to navigate without names always showing.

Ah, right. It could be optional, though.

> Maybe I could experiment with placing the names inside the thumbnail boxes.

Sounds good.

**Comment by StrangeSeasOfThought** [comment]
Perhaps like this:

![compactartistlist2](https://user-images.githubusercontent.com/43582772/64962474-cfd4c580-d897-11e9-9ac3-d1508f0f326f.png)

**Comment by Taiko2k** [comment]
Okay, I've pushed an attempted to implement that concept

**Comment by StrangeSeasOfThought** [comment]
> Okay, I've pushed an attempted to implement that concept

I quite like it! Any chance for displaying the same first two letters only once (like in the mock-up, I'm not a fan of duplicate content...)?

I'll do some more extensive testing in a couple of hours. zZzZz

**Comment by StrangeSeasOfThought** [comment]
Oh, wow, something that immediately comes to mind now:

Clicking on artist image, removes all other artist images and lists all albums below as well as a return button/arrow.

![albums](https://user-images.githubusercontent.com/43582772/65023504-3fe45980-d933-11e9-89a9-ea82a4ff2770.png)

Not really sure if needed, but it looks great.

**Comment by Taiko2k** [comment]
> Removes all other artist images and lists all albums below

Getting a little complicated, and indeed the gallery does kind of fulfill this function. Typically a playlist will have an artists albums grouped together already.

> Ah, I found out about Tab to gallery only now

Fastest way to enter the gallery is using the side button on a mouse if you have one with those.

> Left click artwork to immediately show in playlist would be nice when single click to play is disabled.

It already does that, no?

**Comment by StrangeSeasOfThought** [comment]
> > Ah, I found out about Tab to gallery only now
> 
> Fastest way to enter the gallery is using the side button on a mouse if you have one with those.

Hnngh, indeed.

> > Left click artwork to immediately show in playlist would be nice when single click to play is disabled.
> 
> It already does that, no?

No, after left clicking on album artwork I still have to close the gallery.

**Comment by Taiko2k** [comment]
> No, after left-clicking on album artwork I still have to 'close' the gallery manually. Right click -> Show in Playlist closes the gallery.

Oh, I see what you mean, yes this its intended behavior. (Only when you click show in playlist when in compact mode.) Though also in compact mode, you can click in the far left to exit the gallery.

**Comment by Taiko2k** [comment]
Now I'm being haunted by issues from the past :ghost:

**Comment by StrangeSeasOfThought** [comment]
Heh, no issue exactly. In fact, it doesn't even concern me because I only listen to a relatively small number of artists.

I like that the compact artist list doesn't have a scroll bar. But I think it might be useful for users with a larger amount of artists to have a faster way of scrolling the list (not counting keyboard shortcuts). 

- Keep middle mouse/scroll wheel button pressed (in compact artist list) to fast scroll. Basically like in a browser:

![scroll wheel](https://user-images.githubusercontent.com/43582772/65590525-3b4f1f00-df8b-11e9-98c4-8d1d845cc734.png)

**Comment by Taiko2k** [comment]
That's an idea. I haven't though that one. 

Though.. not having thought of it might be a bad indication, plus being a dated model of scrolling, and not using it anywhere else in the UI, I doubt anyone would think to try scroll that way.

**Comment by StrangeSeasOfThought** [comment]
Yeah, makes sense. I don't know if a faster way of scrolling is even needed.
